application/ledger_query_service/src/lib.rs

```diff
@@ -23,47 +23,27 @@ pub use crate::{
     query_result_repository::*, route_factory::*, routes::*,
 };
 pub use ethereum_support::web3;
-use futures::Future;
-use std::{fmt::Debug, sync::Arc};
+use std::{cmp::Ordering, fmt::Debug, sync::Arc};
 
-type QueryId = u32;
-type QueryMatch = (QueryId, String);
-type ArcQueryRepository<Q> = Arc<dyn QueryRepository<Q>>;
-type ArcQueryResultRepository<Q> = Arc<dyn QueryResultRepository<Q>>;
-
-pub trait BlockProcessor<B> {
-    fn process(
-        &mut self,
-        block: B,
-    ) -> Box<dyn Future<Item = (Vec<QueryMatch>, Vec<QueryMatch>), Error = ()> + Send>;
-}
+#[derive(PartialEq)]
+pub struct QueryId(pub u32);
+#[derive(PartialEq)]
+pub struct QueryMatch(pub QueryId, pub String);
 
-pub trait Query<O>: Debug + 'static {
-    fn matches(&self, object: &O) -> Box<dyn Future<Item = QueryMatchResult, Error = ()> + Send>;
-}
+type ArcQueryRepository<Q> = Arc<dyn QueryRepository<Q>>;
 
 pub trait IsEmpty: Debug + 'static {
     fn is_empty(&self) -> bool;
 }
 
-#[derive(Debug, PartialEq)]
-pub enum QueryMatchResult {
-    Yes { confirmations_needed: u32 },
-    No,
+impl From<u32> for QueryId {
+    fn from(item: u32) -> Self {
+        Self(item)
+    }
 }
 
-impl QueryMatchResult {
-    pub fn yes() -> Self {
-        QueryMatchResult::Yes {
-            confirmations_needed: 0,
-        }
-    }
-    pub fn yes_with_confirmations(confirmations_needed: u32) -> Self {
-        QueryMatchResult::Yes {
-            confirmations_needed,
-        }
-    }
-    pub fn no() -> Self {
-        QueryMatchResult::No
+impl PartialOrd for QueryMatch {
+    fn partial_cmp(&self, other: &QueryMatch) -> Option<Ordering> {
+        Some((self.0).0.cmp(&(other.0).0))
```

Comment:
:wrench: Maybe we can derive `PartialOrd` on `QueryId`? Then we don't have to do this nesting here (so much).